uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -212,8 +213,11 @@ async def asgi_send(self, message):
                 bytes_data = message.get("bytes")
                 text_data = message.get("text")
                 data = text_data if bytes_data is None else bytes_data
-                await self.send(data)
-
+                try:
+                    await self.send(data)
+                except ConnectionClosedError as exc:
```

Comment:
Catching the exception is certainly one way to deal with this. Just want to mention that the `wsproto` implementation checks for `self.transport.is_closing()`:

https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/websockets/wsproto_impl.py#L278

Maybe this is more agnostic to the underlying library. On the other hand it's circumventing the library and checks transport conditions directly. Unsure.

Reply:
yes, I dont like it too, I didn't see something satisfying enough from the faq here too.
https://websockets.readthedocs.io/en/stable/faq.html#what-does-connectionclosederror-code-1006-mean